content/projects/nodejs/api/_index.md

```diff
@@ -30,14 +30,13 @@ There is no need to create a new git repo for this code submission. This is a co
 
 Use Express to expose the following JSON endpoints.
 
-- `/addNewVisitor`: create a new Visitor in the database
-- `/deleteVisitor:id`: delete a single Visitor from the database
-- `/deleteAllVisitors`: delete all Visitors
-- `/viewVisitors`: view all Visitors
-- `/viewVisitor:id`: view a single Visitor
-- `/updateVisitor:id`: Update a single Visitor
+- GET `/visitors` : lists all the visitors
+- POST `/visitors` : creates a new visitors
+- GET `/visitors/{id}` gets the details of a specific visitor
+- DELETE `/visitors/{id}` deletes a specific visitor
+- PUT `/visitors/{id}` updates a specific visitor
```

Comment:
Regarding the representation of the id's I think the previous ones were correct way to represent `path parameters`, they were just missing a slash in between(`/:id`). The way you've represented them now matches how `query parameters` are represented. I think they should still be path not query parameters, I don't know what you think?

See [path parameters vs query parameters](https://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api)

Reply:
[This resource](https://www.google.com/search?q=how+to+represent+path+parameters+in+openapi&oq=how+to+represent+path+para&aqs=chrome.2.69i57j33i160l2.9236j0j1&sourceid=chrome&ie=UTF-8#:~:text=Swagger%20supports%20path%20templating%2C%20meaning%20you%20can%20use%20curly%20braces%20%7B%7D%20to%20mark%20parts%20of%20a%20URL%20as%20path%20parameters%3A%20...%20The%20API%20client%20needs%20to%20provide%20appropriate%C2%A0...) says to use curly brackets to represent path parameters so now I'm confusion :sweat_smile:

Reply:
Actually, good point cause I saw recent changes from a learner where they actually wrote the endpoint exactly like this "visitors/{id}" and the point of {id} was to show that a specific id should be put in

Reply:
oh wait, the second link you sent says otherwise